endpoints/test/discovery_generator_test.py

```diff
@@ -473,6 +470,36 @@ def get_airport(self, request):
         }
     }
 
+class DiscoveryUrlGeneratorTest(BaseDiscoveryGeneratorTest):
+
+  def testUrlGeneration(self):
+    IATA_RESOURCE = resource_container.ResourceContainer(
+        iata=messages.StringField(1)
+    )
+
+    class IataParam(messages.Message):
+        iata = messages.StringField(1)
+
+    class Airport(messages.Message):
+        iata = messages.StringField(1, required=True)
+        name = messages.StringField(2, required=True)
+
+    @api_config.api(name='iata', version='v1')
+    class IataApi(remote.Service):
+        @api_config.method(
+            IATA_RESOURCE,
+            Airport,
+            path='airport/{iata}',
+            http_method='GET',
+            name='get_airport')
+        def get_airport(self, request):
+            return Airport(iata=request.iata, name='irrelevant')
+
+    doc = self.generator.get_discovery_doc([IataApi], hostname='iata.appspot.com')
+    assert doc['baseUrl'] == 'https://iata.appspot.com/_ah/api/iata/v1/'
+    assert doc['rootUrl'] == 'https://iata.appspot.com/_ah/api/'
+    assert doc['servicePath'] == 'iata/v1/'
```

Comment:
Maybe use `self.assertEqual()`?

Reply:
I'm actually switching the code base over to using this style of asserts.

Historically, Python testing has avoided using the `assert` statement in favor of `assertEqual`, `assertNotNone`, `assertSameContents` and so on.

The reason for this is that in normal Python operation, an assert with a condition that evaluates to False simply throws an exception without any details about what went wrong.

However, when we run tests with py.test (and that is the only supported way to run tests right now), py.test detects these assertion errors and inspects the code to produce a rich error message. This means `assert foo == bar` actually produces more useful output than `assertEqual(foo, bar)` in most cases; for instance, if you assert one dict being equal to another, it will show exactly which keys have differences.

Reply:
Awesome, thanks for the explanation!